docs/flare_security_overview.rst

```diff
@@ -0,0 +1,393 @@
+
+- Allow clients to reach the server (directly or via proxies)
+- Maintain confidentiality and integrity of all messages
+- Enforce explicit message authentication to prevent unauthorized access
+
+Explicit Message Authentication
+--------------------------------
+
+FLARE requires every message to include a valid authentication token and signature:
+
+1. Upon startup, the client logs in to the server using credentials from its startup kit
+2. The server issues a token and signature (signed with its private key) binding the client identity to the token
+3. The client attaches its name, token, and signature to each message header
+4. The server validates these before processing the message
+
+This mechanism works independently of TLS or mTLS and relies on PKI credentials in the startup kits. All sites must protect these kits and never share tokens or signatures.
+
+Message authentication applies to all messages—including those from FL clients, 3rd-party systems, and the server itself.
+
+Connection Security Modes
+--------------------------
+
+FLARE's Provision system supports multiple connection security modes, allowing each site to explicitly define how communication is secured.
+
+**TLS (One-way Authentication)**
+
+In TLS, the client authenticates the server using a trusted Root Certificate. The server presents proof signed with its private key, which the client validates using the server's public key. The server does not authenticate the client.
+
+**mTLS (Mutual Authentication)**
+
+In mTLS, both the client and server authenticate each other using PKI credentials from their startup kits. Each side proves its identity by presenting evidence signed with its private key and validating the peer's certificate.
+
+Both parties must use the same mode (TLS or mTLS) for a secure connection.
+
+**Clear Mode**
+
+Clear mode allows unencrypted communication, typically only used when a secure proxy terminates TLS before forwarding traffic to the server.
+
+**Custom Root Certificates**
+
+A custom CA certificate can be used to validate servers in custom network environments, such as those involving intermediate proxies. If not provided, the default root CA generated by the Provision system is used.
+
+If no security mode is specified, mTLS is applied by default.
+
+
+Secure Message Serialization
+============================
+
+In distributed systems, message serialization is a critical security concern. The commonly used Python Pickle mechanism is considered insecure because it can execute arbitrary code during deserialization.
+
+To address this, NVFLARE employs a secure serialization framework called FOBS (FLARE Object Serializer) for all server–client data exchanges.
+
+FOBS Overview
+-------------
+
+FOBS is a secure, Pickle-compatible replacement built on MessagePack. It ensures that only explicitly supported and registered object types can be serialized, preventing code execution or tampering attacks.
```

Comment:
what does pickle compatible mean?

Reply:
true, let me change to pickle replacement